src/surrogates/inc/InterpolationSurrogateDataSet.h

```diff
@@ -0,0 +1,89 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// QUESO - a library to support the Quantification of Uncertainty
+// for Estimation, Simulation and Optimization
+//
+// Copyright (C) 2008-2015 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#ifndef UQ_INTERPOLATION_SURROGATE_DATA_SET_H
+#define UQ_INTERPOLATION_SURROGATE_DATA_SET_H
+
+#include <queso/InterpolationSurrogateData.h>
+
+namespace QUESO
+{
+  class GslVector;
+  class GslMatrix;
+
+  //! Container class for multiple, consistent InterpolationSurrogateData objects
+  /*! This is mostly a convenience object to build and hold multiple
+      InterpolationSurrogateData objects that will have the domian and
+      n_points in each dimension. This is useful for building the surrogates
+      using the InterpolationSurrogateBuilder. */
+  template<class V = GslVector, class M = GslMatrix>
+  class InterpolationSurrogateDataSet
+  {
+  public:
+
+    InterpolationSurrogateDataSet( const BoxSubset<V,M>& domain,
+                                   const std::vector<unsigned int>& n_points,
+                                   unsigned int n_datasets );
+
+    ~InterpolationSurrogateDataSet();
+
+    const InterpolationSurrogateData<V,M>& get_dataset( unsigned int s ) const;
+
+    InterpolationSurrogateData<V,M>& get_dataset( unsigned int s );
+
+    unsigned int size() const
+    { return m_datasets.size(); }
+
+  private:
+
+    InterpolationSurrogateDataSet();
+
+    //! Data structure to hold all data sets
+    std::vector<InterpolationSurrogateData<V,M>*> m_datasets;
```

Comment:
Unfortunately, [you can't use scoped_ptr in a vector](http://stackoverflow.com/questions/17132938/undefined-reference-while-vector-of-boostscoped-ptr) since it doesn't have copy/move semantics. So, we'd need a `unique_ptr` to be able to use in an `std::vector`, but we don't have a wrapper for that (and it's more involved since we need something to fallback on when it's not available), so I just left it as a raw pointer. Perhaps I should add a `\todo` to flag it when when we move to requiring C++11?